src/data/finmap.lean

```diff
@@ -194,6 +204,23 @@ induction_on s $ λ s, alist.lookup_is_some
 theorem lookup_eq_none {a} {s : finmap β} : lookup a s = none ↔ a ∉ s :=
 induction_on s $ λ s, alist.lookup_eq_none
 
+lemma mem_lookup_iff {f : finmap β} {a : α} {b : β a} :
+  b ∈ f.lookup a ↔ sigma.mk a b ∈ f.entries :=
+by { rcases f with ⟨⟨l⟩, hl⟩, exact list.mem_lookup_iff hl }
+
+/-- A version of `finmap.mem_lookup_iff` with LHS in the simp-normal form. -/
```

Comment:
I've never liked this particular simp-normal form, imo it defeats the purpose of defining `∈` to begin with. Anyways, it's probably best to just have this theorem, and prove the other one via `option.mem_def`.

Reply:
IMHO, it doesn't hurt to have 2 theorems and it's easier to find `mem_lookup_iff` if you don't remember the `simp`-normal form.